PWGJE/Tasks/dijetFinderQA.cxx

```diff
@@ -0,0 +1,381 @@
+      return;
+    }
+
+    std::vector<std::array<double, 4>> jetPtcuts;
+    for (auto& jet : jets) {
+      fillJetPtHistogramsData(jet);
+      jetPtcuts.push_back({jet.px(), jet.py(), jet.pz(), jet.energy()});
+    }
+
+    if (jetPtcuts.size() >= 2) {
+      auto& leading_jet = jetPtcuts[0];
+      bool found_pair = false;
+
+      for (size_t i = 1; i < jetPtcuts.size() && !found_pair; i++) {
+        auto& candidate_jet = jetPtcuts[i];
+        Double_t dphi = fabs(atan2(candidate_jet[1], candidate_jet[0]) - atan2(leading_jet[1], leading_jet[0]));
+        if (dphi > M_PI) {
+          dphi = 2 * M_PI - dphi;
+        }
+        if (dphi > 2 * M_PI / 3) {
+          double E = leading_jet[3] + candidate_jet[3];
+          double px = leading_jet[0] + candidate_jet[0];
+          double py = leading_jet[1] + candidate_jet[1];
+          double pz = leading_jet[2] + candidate_jet[2];
+          double dijet_mass = sqrt(E * E - px * px - py * py - pz * pz);
+          fillMassHistogramsData(dijet_mass);
+          found_pair = true;
+        }
+      }
+    }
+  }
+  PROCESS_SWITCH(DijetFinderQATask, processDijetData, "QA for invariant mass of dijet in data", false);
+
+  using JetMCPTable = soa::Filtered<soa::Join<aod::ChargedMCParticleLevelJets, aod::ChargedMCParticleLevelJetConstituents, aod::ChargedMCParticleLevelJetsMatchedToChargedMCDetectorLevelJets>>;
+  void processDijetMCMatched(soa::Filtered<aod::JetCollisionsMCD>::iterator const& collision,
+                             soa::Filtered<soa::Join<aod::ChargedMCDetectorLevelJets, aod::ChargedMCDetectorLevelJetConstituents, aod::ChargedMCDetectorLevelJetsMatchedToChargedMCParticleLevelJets>> const& mcdjets,
+                             JetMCPTable const&, aod::JetTracks const&, aod::JetParticles const&)
+  {
+    if (!jetderiveddatautilities::selectCollision(collision, eventSelection)) {
+      return;
+    }
+
+    std::vector<std::array<double, 4>> jetPtcuts_D;
+    std::vector<std::array<double, 4>> jetPtcuts_P;
+
+    for (auto& jet : mcdjets) {
+      if (jet.has_matchedJetGeo()) {
+        for (auto& matchedJet : jet.template matchedJetPt_as<JetMCPTable>()) {
+          if (matchedJet.pt() > setJetPtCut) {
+            jetPtcuts_D.push_back({jet.px(), jet.py(), jet.pz(), jet.energy()});
+            jetPtcuts_P.push_back({matchedJet.px(), matchedJet.py(), matchedJet.pz(), matchedJet.energy()});
+            break;
+          }
+        }
+      }
+    }
+
+    if (jetPtcuts_D.size() >= 2 && jetPtcuts_P.size() >= 2) {
+      auto& leading_jet_D = jetPtcuts_D[0];
+      auto& leading_jet_P = jetPtcuts_P[0];
+      bool found_pair = false;
+
+      for (size_t i = 1; i < jetPtcuts_D.size() && !found_pair; i++) {
+        auto& candidate_jet_D = jetPtcuts_D[i];
+        auto& candidate_jet_P = jetPtcuts_P[i];
+
+        Double_t dphi_D = fabs(atan2(candidate_jet_D[1], candidate_jet_D[0]) - atan2(leading_jet_D[1], leading_jet_D[0]));
```

Comment:
please use the function constrain angle from the recodecay class that does this phi shift